utils/managed_bytes.hh

```diff
@@ -356,6 +357,14 @@ public:
         }
     }
 
+    // Conforms to FragmentRange<managed_bytes> concept
+    using fragment_type = bytes_view;
+    using fragment_iterator = managed_bytes_view_fragment_iterator;
+    using const_iterator = fragment_iterator;
+    const_iterator begin() const noexcept;
+    const_iterator end() const noexcept;
+    size_t size_bytes() const noexcept { return size(); }
```

Comment:
managed_bytes::size() returns the number of bytes, not the number of fragments. It's wrong to make managed_bytes a FragmentRange, the users expect it to behave as a container of bytes rather than a container of fragments. Need an adapter class for that.